tools/clang/unittests/HLSLTestLib/FileCheckerTest.cpp

```diff
@@ -518,12 +518,32 @@ FileRunCommandPart::RunDxc(dxc::DxcDllSupport &DllSupport,
     if (RequiredDxilMinor != 0xF && stage.compare("rootsig") != 0) {
       // Convert stage to minimum dxil/validator version:
       RequiredDxilMajor = std::max(RequiredDxilMajor, (unsigned)6) - 5;
+
+      bool bInternalValidator =
+          opts.SelectValidator == hlsl::options::ValidatorSelection::Internal;
+      bool bValVerExplicit = opts.ValVerMajor != UINT_MAX;
+
+      // Normally we must check the validator version as well, but there are
+      // two scenarios where the validator version doesn't need to be checked
+      // against the version based on the shader model:
+      // 1. The test selects internal validator.
+      // 2. The test explicitly requests a specific validator version.
       FileRunCommandResult result =
           CheckDxilVer(DllSupport, RequiredDxilMajor, RequiredDxilMinor,
-                       !opts.DisableValidation);
-      if (result.AbortPipeline) {
+                       !(bInternalValidator || bValVerExplicit));
```

Comment:
Why not check  opts.DisableValidation here along with bInternalValidator || bValVerExplicit?

Reply:
The assumption of the original code that it didn't need to check the validator version when `-Vd` was used was incorrect.

That's because `opts.DisableValidation` (or `-Vd`) doesn't disable the behavior of setting the validator version based on the attached validator.  This is the issue hit by tests.  They had used `-Vd` with a SM 6.8 target and were still run when an older validator was attached, which initialized the validator version to one that didn't support the shader model, causing the tests to fail.

However, these other conditions were not checked.  If you select the internal validator, the attached one is irrelevant.  If you explicitly set the version, it needs to be checked against that instead, which is done next.